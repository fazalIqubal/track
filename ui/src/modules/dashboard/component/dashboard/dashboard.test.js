import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import  AdminDashboard  from './dashboard'
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson} from 'enzyme-to-json';

const tenentlistall={
    data: [{
        name:'tenant',
        company_name:'xcdify'
    }]
  
}
const tenantList={
    data: [{
        name:'tenant',
        company_name:'xcdify'
    }]
  
}
const lineChartdata = {
    categories:[10,20,30],
    series:[1]
}
const membershipPlanList ={
    data: [{
        name:'tenant',
        company_name:'xcdify'
    }]
}

Enzyme.configure({ adapter: new Adapter() })
describe('when page is rendering', () => {
    let dashboardComponent
    const mockDispatch = jest.fn(); 
    beforeEach(() => {
        dashboardComponent = shallow(<AdminDashboard.WrappedComponent tenantListAll ={{ tenentlistall }} lineChartdata={{lineChartdata}} tenantList={{tenantList}} membershipPlanList={{membershipPlanList}} dispatch={mockDispatch}/>)
    })
    afterEach(() => {
        dashboardComponent = null 
    })
    it('should render the form', ()=> {
        expect(shallowToJson(dashboardComponent)).toMatchSnapshot();
    })
    it('should render  the tenant overview cards', ()=> {
        expect(dashboardComponent.find('.tenant-overview-cards').exists()).toEqual(true)
    })

    it('should render the pie chart', () => {
        expect(dashboardComponent.find('.pie-chart').exists()).toEqual(true)
    });

    it('should render the line-chart', () => {
        expect(dashboardComponent.find('.line-chart').exists()).toEqual(true)
    });

    it('should render the membershipPlan container', () => {
        expect(dashboardComponent.find('.membershipPlan-container').exists()).toEqual(true)
    });

    it('should render the tenants list container', () => {
        expect(dashboardComponent.find('.tenants-list-container').exists()).toEqual(true)
    });

    it('should render the heading text', () => {
        expect(dashboardComponent.find('.sprint-name').text()).toEqual('Tenants Overview')
    })
})




