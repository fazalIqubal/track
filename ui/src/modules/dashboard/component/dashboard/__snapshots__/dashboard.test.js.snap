// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when page is rendering should render the form 1`] = `
<Layout
  style={
    Object {
      "minHeight": "100vh",
    }
  }
  theme="light"
>
  <Layout
    className="site-layout"
  >
    <div
      className="site-layout-background"
      style={
        Object {
          "minHeight": 360,
          "padding": 24,
        }
      }
    >
      <div
        className="sprint-name heading"
      >
        Tenants Overview
      </div>
      <Row>
        <Col
          span={24}
        >
          <div
            className="tenant-overview-cards"
            span={18}
          />
        </Col>
      </Row>
      <Row>
        <Col
          className="left-dash-content"
          span={16}
        >
          <Card
            className="line-chart"
          />
        </Col>
        <Col
          className="left-pane"
          span={8}
        >
          <Card
            className="pie-chart"
          />
        </Col>
      </Row>
      <Row>
        <Col
          className="left-dash-content"
          span={16}
        >
          <div
            className="tenants-list-container"
          />
        </Col>
        <Col
          className="left-dash-content"
          span={8}
        >
          <div
            className="membershipPlan-container"
          />
        </Col>
      </Row>
    </div>
  </Layout>
</Layout>
`;
